src/languages/swift.js

```diff
@@ -224,24 +242,46 @@ export default function(hljs) {
     USER_DEFINED_ATTRIBUTE
   ];
 
+  // https://docs.swift.org/swift-book/ReferenceManual/Types.html
   const TYPE = {
-    className: 'type',
-    begin: '\\b[A-Z][\\w\u00C0-\u02B8\']*',
-    relevance: 0
-  };
-  // slightly more special to swift
-  const OPTIONAL_USING_TYPE = {
-    className: 'type',
-    begin: '\\b[A-Z][\\w\u00C0-\u02B8\']*[!?]',
-    relevance: 0
+    begin: lookahead(/\b[A-Z]/),
+    relevance: 0,
+    contains: [
+      { // Common Apple frameworks, for relevance boost
+        className: 'type',
+        begin: concat(/(AV|CA|CF|CG|CI|CL|CM|CN|CT|MK|MP|MTK|MTL|NS|SCN|SK|UI|WK|XC)/, Swift.identifierCharacter, '+')
+      },
+      { // Type identifier
+        className: 'type',
+        begin: Swift.typeIdentifier,
+        relevance: 0
+      },
+      { // Optional type
+        begin: /[?!]+/,
+        relevance: 0
+      },
+      { // Variadic parameter
+        begin: /\.\.\./,
+        relevance: 0
+      },
+      { // Protocol composition
```

Comment:
Why do we need these vs just letting the mode end?  Is it all just in service of protocol composition so that this would work:

```
Booger... & Cat
```

Reply:
Could Cat or (& Cat) just be matched separately as it's own type?  This smells a bit like parsing to me vs highlighting.

Reply:
Do you not want to highlight `?` and `!` as part of the type itself?

Reply:
Ah, this is because of operator conflicts...? I see.  That make sense now...

Reply:
Yes, it's just to parse certain operator characters that are used as punctuation here.

> Do you not want to highlight ? and ! as part of the type itself?

This isn't done by other highlighters. I assume that's because it's only syntactic sugar, and not really part of the type name.

Reply:
> This isn't done by other highlighters.

Reply:
Neither Xcode nor VS Code (TextMate grammar) highlight this. Types can get pretty complex in Swift (see the tests), so it makes sense to only highlight the identifiers themselves, not the punctuation.